frontend/src/App.vue

```diff
@@ -45,13 +62,16 @@ import { useTransferHistory } from '@/stores/transfer-history';
 
 const enableFeedback = process.env.NODE_ENV === 'production' && true;
 
-const configuration = useConfiguration();
-const { configurationLoaded } = useLoadConfiguration(configuration.setChainConfiguration);
+const { setConfiguration } = useConfiguration();
+const { loadConfiguration, configurationLoaded, configurationError } =
+  useLoadConfiguration(setConfiguration);
```

Comment:
One of the reasons why i chose to leave it inside its own composable is due to the many status properties we have following this "init" process.
We currently have `error` & `configurationLoaded` status but it can easily be extended to using the `active` status from `useAsynchronousTask` as well.
This means we might end up with 3 status tracking properties. If we move this to the store we would also have to transfer the logic for managing these status properties there as well. 

One more reason is to keep the store file simple and responsible for purely setting/getting its values instead of worrying about how and where to fetch the config from in addition to handling/managing the error status/cases.
This will also lead to easier testing by separation of responsibility.

I do agree though, that passing the `setConfiguration` function to the `useLoadConfiguration` composable is not the most appropriate approach but as long as it doesn't ruin the reusability aspect (of the composable) i think its not a big problem.